lib/Conversion/TorchToLinalg/TorchToLinalg.cpp

```diff
@@ -1275,6 +1275,89 @@ class ConvertAtenUnsqueezeOp : public OpConversionPattern<AtenUnsqueezeOp> {
 };
 } // namespace
 
+#include <iostream>
+
+namespace {
+class ConvertAtenTransposeIntOp
+    : public OpConversionPattern<AtenTransposeIntOp> {
+public:
+  using OpConversionPattern::OpConversionPattern;
+  LogicalResult
+  matchAndRewrite(AtenTransposeIntOp op, llvm::ArrayRef<Value> operands,
+                  ConversionPatternRewriter &rewriter) const override {
+    if (failed(verifyLinalgCompatibleTypes(op, rewriter)))
+      return failure();
+    AtenTransposeIntOp::Adaptor adaptor(operands);
+
+    int64_t dim0;
+    if (!matchPattern(op.dim0(), m_TorchConstantInt(&dim0)))
+      return rewriter.notifyMatchFailure(op, "dim0 must be constant");
+    int64_t dim1;
+    if (!matchPattern(op.dim1(), m_TorchConstantInt(&dim1)))
+      return rewriter.notifyMatchFailure(op, "dim1 must be constant");
+
+    auto inVector = adaptor.self();
+    auto inType = inVector.getType().cast<RankedTensorType>();
+    auto inputRank = inType.getRank();
+    auto outType = getTypeConverter()
+                       ->convertType(op->getResult(0).getType())
+                       .cast<RankedTensorType>();
+    auto elementType = inType.getElementType();
+
+    if (dim0 < 0)
+      dim0 += inputRank + 1;
+    if (dim0 < 0 || dim0 >= inputRank)
+      return rewriter.notifyMatchFailure(op, "dim0 out of range");
+    if (dim1 < 0)
+      dim1 += inputRank + 1;
+    if (dim1 < 0 || dim1 >= inputRank)
+      return rewriter.notifyMatchFailure(op, "dim1 out of range");
+
+    auto loc = op.getLoc();
+
+    llvm::SmallVector<Value> outputDims;
+    for (auto i = 0; i < inputRank; i++) {
+      outputDims.push_back(getDimOp(rewriter, loc, adaptor.self(), i));
+    }
+    auto tmp = outputDims[dim0];
+    outputDims[dim0] = outputDims[dim1];
+    outputDims[dim1] = tmp;
+
+    Value outVector =
+        rewriter.create<linalg::InitTensorOp>(loc, outputDims, elementType);
+    SmallVector<AffineExpr> idExprs;
+    SmallVector<AffineExpr> swapExprs;
+    for (auto i = 0; i < inputRank; i++) {
+      idExprs.push_back(getAffineDimExpr(i, rewriter.getContext()));
+    }
+    for (auto i = 0; i < inputRank; i++) {
```

Comment:
nit: might want to remove the braces for single line if blocks to make the code look cleaner here.  https://llvm.org/docs/CodingStandards.html#don-t-use-braces-on-simple-single-statement-bodies-of-if-else-loop-statements.